File: src/c++/main/commands/RichJson_this.cpp
#include "RichJson_this.hpp"
#include "RichJson_ref.hpp"
#include "../core/RichJsonContext.hpp"

namespace RichJson {

json richJson_this(RichJsonParser& parser, RichJsonContext& context) {
    if (context.currentMember.is_null() ||
        (context.currentMember.is_string() &&
         (context.currentMember.get<std::string>().empty() || context.currentMember.get<std::string>() == "this"))) {
        return *context.current;
    }

    return richJson_resolvePathFrom(context.current, context.containerAddress, parser, context);
}

} // namespace RichJson

File: src/java/main/io/github/_0xmaxlab/commands/RichJson_this.java
package io.github._0xmaxlab.commands;

import io.github._0xmaxlab.core.RichJsonContext;
import io.github._0xmaxlab.core.RichJsonParser;

public class RichJson_this extends RichJson_ref {

    @Override
    public Object execute(RichJsonParser parser, RichJsonContext context) throws Exception {
        if (context.currentMember == null || context.currentMember.toString().isEmpty() || context.currentMember.equals("this")) {
            return context.current;
        }

        return resolvePathFrom(context.current, parser, context);
    }
}